internal/tiltfile/tiltfile_test.go

```diff
@@ -4381,23 +4381,81 @@ allow_k8s_contexts("hello")
 	}
 }
 
-// TODO(dmiller): right now this only tests that `object` is a valid param
-// in the future it will test more
-func TestK8sResourceObjects(t *testing.T) {
+func TestK8sResourceObjectsIncludesNonWorkload(t *testing.T) {
 	f := newFixture(t)
 	defer f.TearDown()
 
 	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("baz"))
 
 	f.file("Tiltfile", `
 docker_build('gcr.io/foo', 'foo')
 k8s_yaml('foo.yaml')
-k8s_resource('foo', objects=['bar'])
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['bar', 'baz:namespace:default'])
 `)
 
 	f.load()
 
-	f.assertNextManifest("foo")
+	f.assertNextManifest("foo", deployment("foo"), k8sObject("bar", "Secret"), k8sObject("baz", "Namespace"))
+	f.assertNoMoreManifests()
+}
+
+func TestK8sResourceObjectsWithSameName(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("bar"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_yaml('namespace.yaml')
+k8s_resource('foo', objects=['bar', 'bar:namespace:default'])
+`)
+
+	f.loadErrString("Found multiple (2) matches for selector")
+}
+
+func TestK8sResourceObjectsCantIncludeSameObjectTwice(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.setupFoo()
+	f.yaml("secret.yaml", secret("bar"))
+	f.yaml("namespace.yaml", namespace("bar"))
+
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', 'foo')
+k8s_yaml('foo.yaml')
+k8s_yaml('secret.yaml')
+k8s_resource('foo', objects=['bar', 'bar:secret:default'])
+`)
+
+	f.loadErrString("Unable to find any unresourced matches for selector bar:secret:default. All YAML has already been resourced")
```

Comment:
I think this error makes sense if someone has a mental model of resource assembly, but is kind of confusing when we present k8s_resource as declarative, not imperative. Also, "All YAML has already been resourced" seems hard to understand.

It'd take some extra bookkeeping, but I think the ideal error message here would be something like "selector %s on resource %s and selector %s on resource %s both match object %s. An object can only be claimed by one selector."

A downside of that is it's pretty special-cased to two selectors matching an object, and doesn't cover two other cases, and we'd need to handle those separately:
* no object exists at all that matches that selector
* the object that matches that selector was already automatically put into a resource (e.g., it's a workload, or a service that matches a workload)

I'd still kind of prefer to generate that error message, but if it's a pain to do all those cases, I think generating something like this would be relatively cheap and cover all cases:

`"Unable to find any uncategorized objects matching selector %s. Did find matching categorized objects %v"`
(where the format of %v is TBD. If there's only one, something like `"bar:secret:default, belonging to resource %s"`. If there are multiple, maybe it should be multiple lines?)

(which will hopefully be easier if we're matching against categorized objects, per previous comment)